orc/src/main/java/org/apache/iceberg/orc/OrcFileAppender.java

```diff
@@ -101,7 +101,7 @@ public Metrics metrics() {
   public long length() {
     Preconditions.checkState(isClosed,
         "Cannot return length while appending to an open file.");
-    return writer.getRawDataSize();
```

Comment:
Just want to note here that although we do check `length` while writing to choose whether to close and start a new file, that doesn't happen for ORC already so it is fine to use a FS call in this method.